devito/ir/stree/algorithms.py

```diff
@@ -109,22 +114,30 @@ def stree_build(clusters, profiler=None, **kwargs):
            any(i.is_Section for i in reversed(tip.ancestors)):
             continue
 
-        candidate = None
-        for i in reversed(tip.ancestors + (tip,)):
+        candidates = tuple(reversed(tip.ancestors[1:] + (tip,)))
+
+        if not any(i.is_Iteration and i.dim.is_Time for i in candidates) and \
+           not candidates[-1] is stree:
+            attach_section(candidates[-1])
+            continue
+
+        found = None
+        for i in candidates:
```

Comment:
it's because it's unstructured by nature , not sure what kind of refactoring I should do to mitigate complexity? any idea? I'll give it another think with a fresh mind this week